js/src/common/components/Tooltip.tsx

```diff
@@ -0,0 +1,207 @@
+   */
+  showOnFocus?: boolean;
+  /**
+   * Tooltip position around element.
+   *
+   * Default: `'top'`.
+   */
+  position?: 'top' | 'bottom' | 'left' | 'right';
+  /**
+   * Whether HTML content is allowed in the tooltip.
+   *
+   * **Warning:** this is a possible XSS attack vector. This option shouldn't
+   * be used wherever possible, and will not work when we migrate to CSS-only
+   * tooltips.
+   *
+   * Default: `false`.
+   *
+   * @deprecated
+   */
+  html?: boolean;
+  /**
+   * Sets the delay between a trigger state occurring and the tooltip appearing
+   * on-screen.
+   *
+   * **Warning:** this option will be removed when we switch to CSS-only
+   * tooltips.
+   *
+   * Default: `0`.
+   *
+   * @deprecated
+   */
+  delay?: number;
+}
+
+/**
+ * The `Tooltip` component is used to create a tooltip for an element. It
+ * surrounds it with a div (or span) which has the required tooltip setup
+ * applied.
+ *
+ * You should use this for any tooltips you create to allow for backwards
+ * compatibility when we switch to pure CSS tooltips instead of Bootstrap
+ * tooltips.
+ *
+ * @example <caption>Basic usage</caption>
+ *          <Tooltip text="You wish!">
+ *            <Button>
+ *              Click for free money!
+ *            </Button>
+ *          </Tooltip>
+ *
+ * @example <caption>Use of `position` and `showOnFocus` attrs</caption>
+ *          <Tooltip text="Woah! That's cool!" position="bottom" showOnFocus>
+ *            <div>3 replies</div>
+ *          </Tooltip>
+ */
+export default class Tooltip extends Component<TooltipAttrs> {
+  private oldText: string = '';
+  private oldVisibility: boolean | undefined;
+
+  private shouldRecreateTooltip: boolean = false;
+  private shouldChangeTooltipVisibility: boolean = false;
+
+  view(vnode) {
+    const { children } = vnode;
+
+    if (this.attrs.title) {
+      console.warn(
+        '`title` attribute was passed to Tooltip component. Was this intentional? Tooltip content should be passed to the `text` attr instead.'
+      );
+    }
+
+    // We remove these to get the remaining attrs to pass to the DOM element
+    const {
+      text,
+      containerType = 'block',
+      tooltipVisible,
+      showOnFocus = true,
+      position = 'top',
+      html = false,
+      delay = 0,
+      className,
+      class: classes,
+      ...attrs
+    } = this.attrs;
+
+    const realText = Array.isArray(text) ? extractText(text) : text;
+
+    // We need to recreate the tooltip if the text has changed
+    if (realText !== this.oldText) {
+      this.oldText = realText;
+      this.shouldRecreateTooltip = true;
+    }
+
+    if (tooltipVisible !== this.oldVisibility) {
+      this.oldVisibility = this.attrs.tooltipVisible;
+      this.shouldChangeTooltipVisibility = true;
+    }
+
+    return (
+      <div title={realText} className={classList('tooltip-container', `tooltip-container--${containerType}`, className, classes)} {...attrs}>
+        {children}
+      </div>
```

Comment:
would it be possible to either directly apply the tooltip logic on the node, or add an `as` prop that allows specifying the tag of the tooltip container, so that in situations such as tooltiping a link, we are not forced to have an html container and therefore not need the `inline-block` and `block` class modifiers, as I can see this as a potential theming issue

Reply:
https://github.com/flarum/core/pull/2843#discussion_r628808845 lol

Reply:
ah.. well, I'd honestly still like to be able to specify the container's HTML tag through a prop, or have the option to not add an HTML container and have the tooltipping instead directly on the node.

As long as that container is not forced and there is a way to not have it and still be able to use tooltips.

Reply:
Sorry, that was a very ambiguous lol. I agree with you on this one. Before other components were changed, it kinda made sense, but looking at it again (especially with class names going on the tooltip :see_no_evil:), I've reverted to https://github.com/flarum/core/pull/2843#pullrequestreview-655105253,